AllJoyn/Platform/Bridge/ScriptAdapterLib/JsAdapter.cpp

```diff
@@ -241,64 +263,70 @@ namespace AdapterLib
         }
 
-        try
-        {
-            // Run the script to get that property. Script function is "setPROPNAME" where PROPNAME is the name of the property
-            // e.g. getBrightness
-
-            AdapterValueVector paramsIn;
-            AdapterValueVector paramsOut;
-            std::string name("set");
-            name += tempProperty->Name();
-            ScriptHost* host = reinterpret_cast<ScriptHost*>((void*)device->HostContext());
-            std::shared_ptr<IAdapterValue> sharedVal(dynamic_cast<IAdapterValue*>(tempValue));
-            paramsIn.push_back(sharedVal);
-
-            host->CallScriptFunction(name, paramsIn, paramsOut);
-            status = ERROR_SUCCESS;
-        }
-        catch (...)
+        // Run the script to get that property. Script function is "setPROPNAME" where PROPNAME is the name of the property
+        // e.g. getBrightness
+
+        AdapterValueVector paramsIn;
+        AdapterValueVector paramsOut;
+        std::string name("set");
+        name += tempProperty->Name();
+        ScriptHost* host = reinterpret_cast<ScriptHost*>((void*)device->HostContext());
+        std::shared_ptr<IAdapterValue> sharedVal(dynamic_cast<IAdapterValue*>(tempValue));
+        paramsIn.push_back(sharedVal);
+
+        auto adapterRequest = std::make_shared<JsAdapterRequest>();
+        RequestPtr = adapterRequest;
+        host->CallScriptFunction(name, paramsIn, false, [=](uint32_t status, AdapterValueVector outParams)
         {
-            std::cerr << "GetPropertyValue failed on adapter" << std::endl;
-            status = static_cast<uint32_t>(ER_FAIL);
-        }
+            if (status != ERROR_SUCCESS)
+            {
+                std::cerr << "GetPropertyValue failed on adapter" << std::endl;
+            }
 
-        return status;
+            adapterRequest->SetStatus(status);
+        });
+
+        return adapterRequest->Status();
     }
 
     _Use_decl_annotations_
     uint32_t
     JsAdapter::CallMethod(
-		std::shared_ptr<IAdapterDevice> device,
-		std::shared_ptr<IAdapterMethod>& method,
-		std::shared_ptr<IAdapterIoRequest>& /*requestPtr*/
+        std::shared_ptr<IAdapterDevice> device,
+        std::shared_ptr<IAdapterMethod>& method,
+        std::shared_ptr<IAdapterIoRequest>& requestPtr
         )
     {
         uint32_t status = 0;
 
         // Call the script function
-        try
-        {
-            ScriptHost* host = reinterpret_cast<ScriptHost*>((void*)device->HostContext());
+        ScriptHost* host = reinterpret_cast<ScriptHost*>((void*)device->HostContext());
 
-            host->CallScriptFunction(method->Name(), method->InputParams(), method->OutputParams());
-            status = ERROR_SUCCESS;
-        }
-        catch (...)
+        auto adapterRequest = std::make_shared<JsAdapterRequest>();
+        requestPtr = adapterRequest;
+        host->CallScriptFunction(method->Name(), method->InputParams(), false, [=](uint32_t status, AdapterValueVector outParams)
```

Comment:
I realized I have a problem with how CallScriptFunction() handles outParams (return values). I'll update the PR with a fix.